Mathlib/Analysis/NormedSpace/Alternating.lean

```diff
@@ -0,0 +1,556 @@
+one can deduce a bound on `f m₁ - f m₂` using the multilinearity.
+Here, we give a precise but hard to use version.
+See `AlternatingMap.norm_image_sub_le_of_bound` for a less precise but more usable version.
+The bound reads
+`‖f m - f m'‖ ≤
+  C * ‖m 1 - m' 1‖ * max ‖m 2‖ ‖m' 2‖ * max ‖m 3‖ ‖m' 3‖ * ... * max ‖m n‖ ‖m' n‖ + ...`,
+where the other terms in the sum are the same products where `1` is replaced by any `i`. -/
+lemma norm_image_sub_le_of_bound' [DecidableEq ι] {C : ℝ} (hC : 0 ≤ C)
+    (H : ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) (m₁ m₂ : ι → E) :
+    ‖f m₁ - f m₂‖ ≤ C * ∑ i, ∏ j, if j = i then ‖m₁ i - m₂ i‖ else max ‖m₁ j‖ ‖m₂ j‖ :=
+  f.toMultilinearMap.norm_image_sub_le_of_bound' hC H m₁ m₂
+
+/-- If `f` satisfies a boundedness property around `0`,
+one can deduce a bound on `f m₁ - f m₂` using the multilinearity.
+Here, we give a usable but not very precise version.
+See `AlternatingMap.norm_image_sub_le_of_bound'` for a more precise but less usable version.
+The bound is `‖f m - f m'‖ ≤ C * card ι * ‖m - m'‖ * (max ‖m‖ ‖m'‖) ^ (card ι - 1)`. -/
+lemma norm_image_sub_le_of_bound {C : ℝ} (hC : 0 ≤ C)
+    (H : ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) (m₁ m₂ : ι → E) :
+    ‖f m₁ - f m₂‖ ≤ C * (Fintype.card ι) * (max ‖m₁‖ ‖m₂‖) ^ (Fintype.card ι - 1) * ‖m₁ - m₂‖ :=
+  f.toMultilinearMap.norm_image_sub_le_of_bound hC H m₁ m₂
+
+/-- If an alternating map satisfies an inequality `‖f m‖ ≤ C * ∏ i, ‖m i‖`,
+then it is continuous. -/
+theorem continuous_of_bound (C : ℝ) (H : ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) :
+    Continuous f :=
+  f.toMultilinearMap.continuous_of_bound C H
+
+/-- Constructing a continuous alternating map from a alternating map
+satisfying a boundedness condition. -/
+def mkContinuous (C : ℝ) (H : ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) : E [Λ^ι]→L[𝕜] G :=
+  { f with cont := f.continuous_of_bound C H }
+
+@[simp] lemma coe_mk_continuous (C : ℝ) (H : ∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) :
+    (f.mkContinuous C H : (ι → E) → G) = f :=
+  rfl
+
+end AlternatingMap
+
+/-!
+### Continuous alternating maps
+
+We define the norm `‖f‖` of a continuous alternating map `f` in finitely many variables as the
+smallest number such that `‖f m‖ ≤ ‖f‖ * ∏ i, ‖m i‖` for all `m`. We show that this
+defines a normed space structure on `ContinuousMultilinearMap 𝕜 E G`.
+-/
+
+namespace ContinuousAlternatingMap
+
+variable (c : 𝕜) (f g : E [Λ^ι]→L[𝕜] G) (m : ι → E)
+
+theorem bound : ∃ (C : ℝ), 0 < C ∧ (∀ m, ‖f m‖ ≤ C * ∏ i, ‖m i‖) :=
+  f.toContinuousMultilinearMap.bound
+
+instance : NormedAddCommGroup (E [Λ^ι]→L[𝕜] G) :=
+  NormedAddCommGroup.induced _ _
+    (toMultilinearAddHom : E [Λ^ι]→L[𝕜] G →+ _)
+    toContinuousMultilinearMap_injective
+
+@[simp] lemma norm_toContinuousMultilinearMap : ‖f.1‖ = ‖f‖ := rfl
+
+/-- The inclusion of `E [Λ^ι]→L[𝕜] G` into `ContinuousMultilinearMap 𝕜 (fun _ : ι ↦ E) G`
+as a linear isometry. -/
+@[simps!] def toContinuousMultilinearMapLinearIsometry :
+    E [Λ^ι]→L[𝕜] G →ₗᵢ[𝕜] ContinuousMultilinearMap 𝕜 (fun _ : ι ↦ E) G :=
```

Comment:
Does `ContinuousMultilinearMap` have notation you can use here too?

Reply:
No, `ContinuousMultilinearMap` has notation only in the case `ι = Fin n`. We may decide to change this but not in this PR.